ipatests/test_integration/test_user_permissions.py

```diff
@@ -93,3 +93,41 @@ def test_stageuser_show_as_alternate_admin(self):
         # the field Kerberos Keys available must contain True
         result = self.master.run_command(['ipa', 'stageuser-show', stageuser])
         assert 'Kerberos keys available: True' in result.stdout_text
+
+
+class TestInstallClientNoAdmin(IntegrationTest):
+
+    def test_installclient_as_user_admin(self):
+        """ipa-client-install should not use hardcoded admin for principal
+        Related to : https://pagure.io/freeipa/issue/5406
+        """
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        username = 'testuser1'
+        password = self.master.config.admin_password
```

Comment:
Please use a different, unique password for the user to ensure that we are really not hard-coding admin anywhere.